src/read.rs

```diff
@@ -310,6 +311,49 @@ impl<R: Read + io::Seek> ZipArchive<R> {
             comment: footer.zip_file_comment,
         })
     }
+    /// Extract a Zip archive into a directory.
+    ///
+    /// Malformed and malicious paths are rejected so that they cannot escape
+    /// the given directory.
+    ///
+    /// This bails on the first error and does not attempt cleanup.
```

Comment:
Perhaps we should also mention that this will override existing files if they have the same path as a file in the zip archive.